src/bp/ui-shared/src/Contents/Components/Fields/Text.tsx

```diff
@@ -1,24 +1,34 @@
+import { FormField, FormFieldType } from 'botpress/sdk'
 import React, { FC, useEffect, useState } from 'react'
 
+import { getFieldDefaultValue } from '../../utils/fields'
 import style from '../style.scss'
 import { FieldProps } from '../typings'
 
 interface TextProps extends FieldProps {
-  type: string
+  type: FormFieldType
 }
 
 const Text: FC<TextProps> = ({ onBlur, onChange, placeholder, type, value }) => {
-  const [localValue, setLocalValue] = useState(value || '')
+  const [localValue, setLocalValue] = useState(value || getFieldDefaultValue({ type } as FormField))
 
   useEffect(() => {
-    setLocalValue(value || '')
+    setLocalValue(value || getFieldDefaultValue({ type } as FormField))
```

Comment:
Instead of casting each instance, you can change the signature to make it partial:
 `export const getFieldDefaultValue = (field: Partial<FormField>) => {`

Then you can remove all your "as FormField"